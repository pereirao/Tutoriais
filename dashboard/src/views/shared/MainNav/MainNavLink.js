//@flow

import React from 'react';
import {Link} from 'react-router-dom';
import classNames from 'classnames/bind';

import I18n from '../../../localization';
import style_sheet from './MainNav.module.css';
import NotificationCount from './NotificationCount';

const cx = classNames.bind(style_sheet);

type MainNavLinkProps = { current_path: string, route_name: string };
const MainNavLink = ({current_path, route_name}: MainNavLinkProps) => {
  const is_active = current_path.startsWith(`/${route_name}`) ||
    (route_name === 'orders' && current_path.startsWith('/order'));
  return (
    <Link
      to={`/${route_name}`}
      title={I18n.t(`accessibility.title.${route_name}`)}
      className={cx('navLink', {activeTab: is_active})}>
      {I18n.t(`ui.tab.${route_name}`)}
      {route_name === 'orders' && <NotificationCount/>}
    </Link>
  );
};

export default MainNavLink;
