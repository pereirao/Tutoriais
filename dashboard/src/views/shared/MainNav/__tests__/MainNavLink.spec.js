import React from 'react';

import MainNavLink from '../MainNavLink';

it('renders correctly', () => {
  expect(shallow(<MainNavLink route_name="reports" current_path="/orders" />)).toMatchSnapshot();
});
