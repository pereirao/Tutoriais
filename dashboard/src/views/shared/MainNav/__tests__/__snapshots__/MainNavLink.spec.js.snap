// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Link
  className="navLink"
  replace={false}
  title="Create and View Reports"
  to="/reports"
>
  REPORTS
</Link>
`;
